dali/kernels/common/scatter_gather.h

```diff
@@ -77,16 +77,23 @@ class DLL_PUBLIC ScatterGatherGPU {
     }
   }
 
+  enum class CopyMethod {
+    Default = 0,  // Uses scatter-gather kernel, unless there are 2 or fewer single effective copy
+                  // ranges, in that cases cudaMemcpyAsync is used
+    MemcpyAlways = 1,  // Always use cudaMemcpyAsync
+    MemcpyNever = 2,   // Always use scatter-gather kernel
+  };
```

Comment:
Option 1:
```suggestion
  enum class Method {
    Default = 0,  // Uses scatter-gather kernel, unless there are 2 or fewer single effective copy
                  // ranges, in that cases cudaMemcpyAsync is used
    Memcpy = 1,  // Always use cudaMemcpyAsync
    Kernel = 2,   // Always use scatter-gather kernel
  };
```
This reduces
`ScatterGather::CopyMethod::MemcpyAlways`
to
`ScatterGather::Method::Memcpy`

Option 2:
```suggestion
  enum Method {
    MethodDefault = 0,  // Uses scatter-gather kernel, unless there are 2 or fewer single effective copy
                  // ranges, in that cases cudaMemcpyAsync is used
    MethodMemcpy = 1,  // Always use cudaMemcpyAsync
    MethodKernel = 2,   // Always use scatter-gather kernel
  };
```
`ScatterGather::MethodMemcpy` (even shorter than previous one).